src/main/java/org/erp/school/client/controller/ClientController.java

```diff
@@ -2,46 +2,53 @@
 
 import io.swagger.annotations.Api;
 import lombok.extern.slf4j.Slf4j;
-import org.erp.school.document.dto.DocumentDto;
-import org.erp.school.document.service.DocumentService;
-import org.springframework.http.HttpStatus;
+import org.erp.school.client.dto.ClientDTO;
+import org.erp.school.client.services.ClientService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import javax.validation.Valid;
-import java.util.List;
-import java.util.stream.Collectors;
 
 @RestController
-@Api
 @RequestMapping("/client")
+@Api
 @Slf4j
 public class ClientController {
 
-  private final DocumentService documentService;
+  private final ClientService clientService;
+
+  public ClientController(ClientService clientService) {
+    this.clientService = clientService;
+  }
+
+  @GetMapping
+  public ResponseEntity<Page<ClientDTO>> getAllClients(Pageable pageable) {
+    return ResponseEntity.ok(clientService.getAllClients(pageable));
+  }
+
+  @GetMapping("/{id}")
+  public ResponseEntity<ClientDTO> getClient(@PathVariable String id) {
+    return ResponseEntity.ok(clientService.getClient(id));
+  }
+
+  @PostMapping
+  public ResponseEntity<Void> saveClient(@RequestBody @Valid ClientDTO dto) {
+    return ResponseEntity.created(clientService.saveClient(dto)).build();
+  }
 
-  public ClientController(DocumentService documentService) {
-    this.documentService = documentService;
+  @PutMapping("/{id}")
+  public ResponseEntity<Void> updateClient(
+      @RequestBody @Valid ClientDTO dto, @PathVariable String id) {
+    dto.setId(id);
```

Comment:
this is dangerous. I would have a validation that the dto.id is the same as the idfrom the parameter.

Reply:
How is this dangerous? I mean I can change it no problem, I just use the same approach in some other projects, if this is really dangerous I would like to know what threat it has.

Reply:
It is just for a validation that the `dto` from your request is not the data from another `id`.
E.g.:
dto
|_id = 5
|_data1=*
|_data2=*

But I send the request via /6, so id is 6.

I would just check that the object you receive in the request has the same id as the parameter.